### ISSUE TITLE
AssertionError in LinArSolve.mbtc(...)

### ISSUE THREAD

**Issue by danieldietsch** [ultimate-pa/smtinterpol#37]
SMTInterpol 2.5-40-g69ef504908 (build from 69ef504908) crashes on the [attached input script](https://github.com/ultimate-pa/smtinterpol/files/2274112/rcfgbuilder.zip) with a NullPointerException. When enabling assertions, the crash is an assertion error. 

Stacktrace: 
````
Exception in thread "main" java.lang.AssertionError
        at de.uni_freiburg.informatik.ultimate.smtinterpol.theory.linar.LinArSolve.mbtc(LinArSolve.java:2061)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.theory.linar.LinArSolve.computeConflictClause(LinArSolve.java:618)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.dpll.DPLLEngine.checkConsistency(DPLLEngine.java:1027)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.dpll.DPLLEngine.solve(DPLLEngine.java:1171)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.SMTInterpol$CheckType$1.check(SMTInterpol.java:114)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.SMTInterpol.checkSatAssuming(SMTInterpol.java:502)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.SMTInterpol.checkSat(SMTInterpol.java:455)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.Parser$Action$.CUP$do_action(Parser.java:2279)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.Parser.do_action(Parser.java:1146)
        at com.github.jhoenicke.javacup.runtime.LRParser.parse(Unknown Source)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.ParseEnvironment.parseStream(ParseEnvironment.java:147)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.ParseEnvironment.parseScript(ParseEnvironment.java:125)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.smtlib2.SMTLIB2Parser.run(SMTLIB2Parser.java:37)
        at de.uni_freiburg.informatik.ultimate.smtinterpol.Main.main(Main.java:180)
````

**Comment by jhoenicke** [comment]
Minimized script:

```
(set-logic QF_AUFLIRA)
(declare-fun req1_Glob_BndResponseUT_1_X2_In164_const_-317914485 () Real)
(declare-fun req1_Glob_BndResponseUT_1_X2_Out164_const_-1020592248 () Real)
(declare-fun req1_Glob_BndResponseUT_1_pc_In164_const_-900185023 () Int)
(declare-fun req1_Glob_BndResponseUT_1_pc_Out164_const_-1953419470 () Int)
(assert (or (and (not (= req1_Glob_BndResponseUT_1_pc_In164_const_-900185023 0)) (= req1_Glob_BndResponseUT_1_X2_In164_const_-317914485 req1_Glob_BndResponseUT_1_X2_Out164_const_-1020592248) (= req1_Glob_BndResponseUT_1_pc_In164_const_-900185023 req1_Glob_BndResponseUT_1_pc_Out164_const_-1953419470)) (= req1_Glob_BndResponseUT_1_X2_Out164_const_-1020592248 0.0)))
(check-sat)
```

Problem is that we share `0` and `0.0` as different shared terms in the clausifier and that triggers an assertion in mbtc or a NullPointerException.

Possible fix: normalize constants like 0 and 0.0 to the same term (prefer Int or Real?) in `TermCompiler.convert()`.

**Comment by jhoenicke** [comment]
Or we could think of 0 and 0.0 as different and not merge SharedTerms that have different type.
